### PR TITLE
Done

### PR DESCRIPTION
**Pull request by vovkandrew**

### CODE REVIEW

File: src/main/java/core/basesyntax/BooleanExpression.java [hunk 1]
```diff
@@ -13,7 +13,9 @@
      * <p>Это задание необходимо решать с использованием логических операторов,
      * без использования конструкций if-else</p>
      */
+
     public boolean booleanExpression(boolean a, boolean b, boolean c, boolean d) {
-        return false;
+        return ((a && b) && (!c && !d)) || ((a && c) && (!b && !d)) || ((a && d) && (!c & !b))
+                || ((b && c) && (!a && !d)) || ((b && d) && (!a && !c)) || ((c && d) && (!a && !b));
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/BooleanExpression.java]: try to simplify it with binary operators

Comment (vovkandrew) [reply src/main/java/core/basesyntax/BooleanExpression.java]: Simplified

File: src/main/java/core/basesyntax/BooleanExpression.java [hunk 2]
```diff
@@ -13,7 +13,8 @@
      * <p>Это задание необходимо решать с использованием логических операторов,
      * без использования конструкций if-else</p>
      */
+
     public boolean booleanExpression(boolean a, boolean b, boolean c, boolean d) {
-        return false;
+        return ((a ^ b) & (c ^ d)) | ((a ^ c) & (b ^ d)) | ((a ^ d) & (c ^ b));
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/BooleanExpression.java]: there are redundant checks here, try to simplify it

Comment (vovkandrew) [reply src/main/java/core/basesyntax/BooleanExpression.java]: Don't see any
How is that possible if I check all the possible combinations? Or where is redundancy?

Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/BooleanExpression.java]: you should check all unique cases, there are some cases that is cover by others(so they are redundant)

**Review by dmytro-hryhoruk — APPROVED** [review]
Good job!

**Pull request closed without merging**